subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.api.provider.ProviderFactory;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.process.internal.ExecHandleBuilder;
+import org.gradle.process.internal.ExecHandleFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class OsXInstallationSupplier extends AutoDetectingInstallationSupplier {
+
+    private final ExecHandleFactory execHandleFactory;
+    private final OperatingSystem os;
+
+    public OsXInstallationSupplier(ExecHandleFactory execHandleFactory, ProviderFactory providerFactory, OperatingSystem os) {
+        super(providerFactory);
+        this.execHandleFactory = execHandleFactory;
+        this.os = os;
+    }
+
+    @Override
+    protected Set<InstallationLocation> findCandidates() {
+        if (os.isMacOsX()) {
+            final Reader output = executeJavaHome();
+            final Set<File> javaHomes = new OsXJavaHomeOutputParser().parse(output);
+            return javaHomes.stream().map(this::asInstallation).collect(Collectors.toSet());
+        }
+        return Collections.emptySet();
+    }
+
+    private InstallationLocation asInstallation(File javaHome) {
+        return new InstallationLocation(javaHome, "macOS java_home");
+    }
+
+    private Reader executeJavaHome() {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        executeCommand(outputStream);
+        return new InputStreamReader(new ByteArrayInputStream(outputStream.toByteArray()));
+    }
+
+    void executeCommand(ByteArrayOutputStream outputStream) {
+        ExecHandleBuilder execHandleBuilder = execHandleFactory.newExec();
+        execHandleBuilder.workingDir(new File(".").getAbsoluteFile());
+        execHandleBuilder.commandLine("/usr/libexec/java_home", "-V");
```

Comment:
Why not use the `-X` option that would give you metadata in one go?

Reply:
Was thinking about it. Honestly, in "one go" isn't really what -X is doing. It just gives us a different format (plist formatted as XML). Parsing the XML to grab the `JVMHomePath` elements would simply mean we use some object mapper (jaxb) or try to parse it using jackson+json path. I agree that the `plist+xml` sounds more "stable", not sure it's worth it. WDYT?

Reply:
I mean that you get the vendor, more specific version, architecture, etc ...
Which we may not need now but might in the future. And if we start by leveraging that format, extracting additional fields will be trivial.

Reply:
Since it is not needed right now and this approach has been tested for a while, let's go with it for now.